### ISSUE TITLE
Conversion between 8-bit and 16-bit pixels

### ISSUE THREAD

**Issue by Patiga** [image-rs/image#1367]
I would like to be able to convert 16-bit image types to their 8-bit counterpart and vice versa.

My specific use case for this functionality is loading any png into a struct that requires rgba 8-bit images.
Converting the different image gray-scale, rgb, etc. into rgba can be done via `into_rgba`, however there doesn't seem to be anything for converting the 8-bit/16-bit variants into each other.

This is always applicable when you need to convert a format that only accepts a certain bit-width.
For example saving a bmp or gif with this library requires 8-bit pixels.

## Draft

8-bit to 16-bit could be done by using the 8-bit value twice to fill the 16-bit. This would map min to min (both 0), max to max (both all 1's).
16-bit to 8-bit could be done by using the 8 most significant bits.

**Comment by fseegraeber** [comment]
There now is `to_rgb8` for a `DynamicImage` if that is what you are looking for. Included since release 0.23.12.

I think you might be able to prevent the image copy (if that is an issue), if you use the [png](https://crates.io/crates/png) crate directly. The decoder seems to support directly transforming on decode.

**Comment by 197g** [comment]
Yep, we're supporting native 16-bit encoding and encoding since `0.25` if you go through `DynamicImage`. For color space aware conversion beyond that, see #2531 .
